libraries/rush-lib/src/schemas/version-manifest.schema.json

```diff
@@ -0,0 +1,41 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema",
+  "title": "Version manifest output file format",
+  "description": "A version manifest is an output file produced by the 'rush version' command in Rush, describing the set of projects that have new versions and should be published and/or deployed.",
+  "type": "object",
+  "required": ["projects"],
+  "additionalProperties": false,
+  "properties": {
+    "$schema": {
+      "description": "Part of the JSON Schema standard, this optional keyword declares the URL of the schema that the file conforms to. Editors may download the schema and use it to perform syntax highlighting.",
+      "type": "string"
+    },
+    "projects": {
+      "description": "A list of versioned projects.",
+      "type": "array",
+      "items": {
+        "type": "object",
+        "required": ["name", "relativeFolderPath", "version"],
+        "additionalProperties": false,
+        "properties": {
+          "name": {
```

Comment:
Unless duplicates are allowed, it makes more semantic sense to have this be a dictionary of package name to change data. I know @octogonz prefers named fields for auto-complete, but this isn't a manually authored file.

Reply:
🤔 I'm fine with either. I think I was trying to optimize for ease-of-use across other tooling, but they might be too similar to pick a favorite.

### Use case 1: some kind of GHA workflow matrix

```console
# Array
echo "FOLDERS=$(cat manifest.json | jq '[.[].relativeFolderPath]' -c)" >> $GITHUB_OUTPUT

# Dictionary
echo "FOLDERS=$(cat manifest.json | jq '[values[].relativeFolderPath]' -c)" >> $GITHUB_OUTPUT
```

### Use case 2: some kind of inline report

In this case I think array is a bit easier but they are comparable:

```console
# Array
node -pe '"### Publishing\n\nName | Version\n--- | ---\n" + require("./manifest.json").map(x => x.name + " | " + x.version).join("\n")'

# Dictionary
node -pe 'a=require("./manifest.json"),"### Publishing\n\nName | Version\n--- | ---\n" + Object.keys(a).map(x => x + " | " + a[x].version).join("\n")'
```

Reply:
For the node example, `Object.entries` will convert it back to an array of `[key, value]` tuples, so:
```
node -pe '`### Publishing\n\nName | Version\n--- | ---\n${Object.entries(require(`./manifest.json`)).map(([name, { version }]) => `${name} | ${version}`).join(`\n`)}`'
```

Pretty negligible complexity difference, but makes the file guaranteed to not contain duplicates.